### ISSUE TITLE
Full versions table is broken on opencv-contrib-python

### ISSUE THREAD

**Issue by waveform80** [piwheels/piwheels#312]
Expand table on https://www.piwheels.org/project/opencv-contrib-python/ to see several broken rows

**Comment by bennuttall** [comment]
I just deleted the builds for a version of a package, and once the cp35m and cp37m builds were completed and the cp39 build was pending, it didn't have a `<td>` for the cp39 column:

![Screenshot from 2022-06-08 23-56-23](https://user-images.githubusercontent.com/1609725/172732987-598ff144-4d0f-4510-a6d2-fd8cfa24d300.png)

**Comment by waveform80** [comment]
Okay, figured part of this out over the weekend; there's a bug in the ABI scoring calculation in `get_project_data` as I'd feared. A couple of example cases:

* In [pika](https://www.piwheels.org/project/pika/) version 1.0.1 has builds for `cp34m` (failed), and `cp37m` (succeeded, produced "none-any" wheel). This means the ABI scoring calculation produces output for `cp37m` and `cp39` (because the wheel is compatible with both) but nothing for `cp35m` because there's a `cp34m` build that the `LEFT JOIN` on `builds` matches to `cp35m` (potentially compatible) but the scoring logic rejects because there's no file output. Hence, the ABI never appears in the output.
* In [opencv-contrib-python](https://www.piwheels.org/project/opencv-contrib-python/) version 4.1.1.26 is skipped (binary only) but has manually imported builds for `cp34m`, `cp35m`, and `cp37m`. The `LEFT JOIN` on `builds` thus produces joined rows with `build_abis` but none with compatible files for `cp39` so that ABI is never seen in the output.

Basically we need to ensure that there's always a mismatched `build` row (all NULLs) for each `build_abis` row. This'll probably need forcing with a `UNION` somewhere ... unless I can come up with something more elegant...
